massa-network-worker/src/node_worker.rs

```diff
@@ -161,14 +117,29 @@ impl NodeWorker {
                     }
                 };
             }
+
+            if let Err(e) = socket_writer.write_half.shutdown().await {
```

Comment:
Why we are shutting down the first writer ? We can't make one single writer for the whole process ?

Reply:
What do you mean by first writer? And not sure to understand your 2nd question too :-/

Reply:
Ok my bad